### PR TITLE
Add support for ruby 2.7

### LINKED ISSUE #437: please add ruby2.7 support
As mentionned in https://github.com/fakefs/fakefs/pull/436, the tests do not pass anymore on ruby2.7.
Causes are:
- tremendous ammount of warning caused by deprecation of taint/tainted?/$SAFE etc. on ruby2.7
  https://bugs.ruby-lang.org/issues/16131
- additional double splatter operators appearing in constructor of `tempfile.rb` of ruby
https://github.com/ruby/ruby/blob/b76a21aa45fff75909a66f8b20fc5856705f7862/lib/tempfile.rb#L127
https://github.com/ruby/ruby/blob/b76a21aa45fff75909a66f8b20fc5856705f7862/lib/tempfile.rb#L129
, which does not work anymore with the `create` method of `lib/fakefs/dir.rb`

Could you please fix this to add ruby2.7 support? Thanks in advance.

### PR DESCRIPTION
**Pull request by terceiro**
Fixes #437

### CODE REVIEW

File: lib/fakefs/dir.rb [hunk 1]
```diff
@@ -194,20 +194,16 @@ def create(basename, *rest)
         if (opts = Hash.try_convert(rest[-1]))
           opts = opts.dup if rest.pop.equal?(opts)
           max_try = opts.delete(:max_try)
-          opts = [opts]
+          opts = opts
```
Comment (grosser) [inline lib/fakefs/dir.rb]: can delete this line I guess

**Comment by grosser** [comment]
looks good!

... what's with the taint logic being removed, did that break/was that not needed on all ruby versions ?

File: lib/fakefs/dir.rb (hunk 1 above)
Comment (terceiro) [reply lib/fakefs/dir.rb]: yes! fixed

**Comment by terceiro** [comment]
> ... what's with the taint logic being removed, did that break/was that not needed on all ruby versions ?

In ruby 2.7, $SAFE, taint, tainted? etc are no-ops, and they will be removed completely in ruby 3.2

See https://bugs.ruby-lang.org/issues/16131 for details

**Comment by grosser** [comment]
v1.1.0

**Pull request merged**